### ISSUE TITLE
ANR / Gateway freezing on Android version 7.0

### ISSUE THREAD

**Issue by rosteve** [medic/medic-gateway#132]
## Device details
-----

# Device tested

key | value
---|---
medic-gateway version | _1.5.0_
device make | _Samsung_
device model | _J5_
can the problem be recreated on this device? | _yes_
Android version | _7.0_

# Observed behaviour

_When the gateway is left unattended for a couple of hours (approx. 3-4 hours), it doesn't respond and any sms report sent to it or received is not forwarded until you force it to refresh or give a miss-call_

Also the phone generally freezes intermittently and becomes unresponsive._

_The gateway becomes responsive when we connect it to the machine or source of power hence its hard to collect the source of the gateway app misbehaving in the logs since we need to connect it to a machine. Also the gateway does not hang when it is persistently connected to the laptop to take logs_.

# Steps we have taken to try to mitigate this
1. We have tried to enable stay awake in developer's mode
2. Removed battery optimization 
3. Changed advanced settings of the phone which did help by extending the time of the gateway freezing by few hours; but if the phone is left undisturbed it freezes. 
4. Taken a couple of logs (some overnight) but the logs seem to to be pretty clean
The above unfortunately did not stop the issue.

# Expected behaviour

_The gateway app should not freeze and should send/receive reports in the background even when the gateway is idling._

# Steps to recreate

1. Log into the gateway (the credentials will be provided upon request)
2. Leave the gateway unattended for few hours and hope that it stops interacting with the WebApp. 

One can connect to the gateway using teamviewer [Note: If the gateway phone is connected using the TeamViewer, it starts responding.

You can get more information on this issue on the `standard-asia` channel, where we've been having discussion around this

cc: @shreyakb @ngamita @mukesh2006 @kanikamedic @alxndrsn

**Comment by alxndrsn** [comment]
@rosteve your first three mitigation steps look like a great checklist of things to do as part of gateway phone setup.

It's worth noting that _Stay Awake_ is only meant to work when the phone is plugged in to external power.

> The gateway becomes responsive when we connect it to the machine or source of power 

@ngamita @derickl & all: Would it be reasonable to recommend that all gateway phones:
1. hava _Developer Options_ > _Stay Awake_ enabled, and
2. stay plugged in to mains power as much as possible

?

I understand that both of these are onerous and don't provide the simplest usability imaginable.  However, from the report above it sounds like they solve the existing problem.   Perhaps we can use them while we search for a long-term solution that is easier for users to implement.

**Comment by alxndrsn** [comment]
Related: https://github.com/medic/medic-gateway/issues/127, https://github.com/medic/medic-gateway/issues/126

**Comment by derickl** [comment]
@alxndrsn that's reasonable given the current situation (it's the simplest fix)

**Comment by derickl** [comment]
@osoro - is downgrading a viable alternative - that is if we haven't had similar issues with Android < 7.0

**Comment by rosteve** [comment]
Yeah, that's that a viable alternative since we noticed its works well with <= 6.0  but @mukesh2006  would be best to answer this

**Comment by mukesh2006** [comment]
@alxndrsn @derickl @ngamita @rosteve We can request the partners to buy a phone with Android <= 6.0 but downgrading might not be a viable option for them at their end. 

As @rosteve already tested it on Android 6.0.1, and the gateway was not freezing, I don't think it's worth downgrading the Android version on the gateway phone which @kanikamedic has.
